src/lib/address/address.ts

```diff
@@ -8,14 +9,45 @@ import {
   validateUnstoppableDomain,
 } from 'lib/address/unstoppable-domains'
 import { resolveYat, validateYat } from 'lib/address/yat'
+import { logger } from 'lib/logger'
 import { store } from 'state/store'
 
 import { ensReverseLookupShim } from './ens'
 
+const moduleLogger = logger.child({ namespace: ['lib', 'address'] })
+
 type VanityAddressValidatorsByChainId = {
   [k: ChainId]: ValidateVanityAddress[]
 }
 
+type ParseUrlByChainId = (chainId: ChainId, url: string) => ParseUrlByChainIdReturn | undefined
+
+type ParseUrlByChainIdReturn = {
+  target_address: string
+  parameters?: {
+    value: string
+    gas: string
+    gasPrice: string
+  }
+}
```

Comment:
```suggestion
type ParseUrlByChainIdReturn = ReturnType<typeof parse>
```

this way it stays up to date with upstream if `parse` ever changes

Reply:
Note, this loses type-checking - `typeof` conceptually makes sense though, sounds like this needs a `definitelytyped` PR and/or properly defining types in the `declare module` directive

With that being said, we do some parsing and don't return the `typeof parse`, so that's incorrect, see:

https://github.com/shapeshift/web/blob/1c25de1d4954877fd8a0c0fb3a80c91fe1fba2a0/src/lib/address/address.ts#L40-L45

The reason why the types work is precisely because we lose type-checking